crates/fhir-sdk/src/client/builder.rs

```diff
@@ -87,8 +96,13 @@ impl<V> ClientBuilder<V> {
 			return Err(Error::UrlCannotBeBase);
 		}
 
-		let user_agent = self.user_agent.as_deref().unwrap_or(DEFAULT_USER_AGENT);
-		let client = reqwest::Client::builder().user_agent(user_agent).build()?;
+		let client = match self.client {
+			Some(client) => client,
+			None => {
+				let user_agent = self.user_agent.as_deref().unwrap_or(DEFAULT_USER_AGENT);
```

Comment:
Could you add documentation to the user agent function that it is ignored if the client is given?